_layouts/single.html

```diff
@@ -17,21 +17,26 @@
 <div id="main" role="main">
   {% include sidebar.html %}
 
-  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
+  <article class="page h-entry" itemscope itemtype="https://schema.org/CreativeWork">
     {% if page.title %}<meta itemprop="headline" content="{{ page.title | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.excerpt %}<meta itemprop="description" content="{{ page.excerpt | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
     {% if page.date %}<meta itemprop="datePublished" content="{{ page.date | date_to_xmlschema }}">{% endif %}
     {% if page.last_modified_at %}<meta itemprop="dateModified" content="{{ page.last_modified_at | date_to_xmlschema }}">{% endif %}
 
+  <ul class="hidden">
+    <li>These list items are microformat entries and are hidden from view.</li>
+    <li class="u-url">{{ page.url | absolute_url }}</li>
+  </ul>
```

Comment:
Hmm—the specification is fuzzy ("[No properties are explicitly required](https://microformats.org/wiki/h-entry#what_is_the_bare_minimum_list_of_required_properties_on_an_h-entry)") and then goes on to say an "h-entry should have the following properties at a minimum for consumers:"

* `url`
* `dt-published` (timestamp)

In this latest commit, I moved `dt-published` to `page__date.html` as you suggested and removed `dt-updated` (the if..else markup in `page__date.html` doesn't distinguish between published and updated when it is rendered to the page).  I also removed `p-author` and `p-summary` as they aren't in the required subset.  

That leaves `u-url`.  Interestingly, there is a [`{{ page.link}}` in `_layouts/single.html`](https://github.com/mmistakes/minimal-mistakes/blob/master/_layouts/single.html#L44), but I can't find a condition where `page.link` would have a value (unless it is expected to come from the page front matter).  Unfortunately, `u-url` is a required data element.  Thoughts on how to handle this?

> I'm still iffy on the duplicating all the author markup as it's in the sidebar with microformat markup now.

The duplication is necessary so the author information occurs within the enclosing `h-entry` markup, and the sidebar is outside that markup.  But since it isn't required I just removed it.

Reply:
> The duplication is necessary so the author information occurs within the enclosing h-entry markup, and the sidebar is outside that markup. But since it isn't required I just removed it.

Now I’m remembering why I wasn’t keen on adding microformats years ago 🤣. It would be so much better if it leveraged other standards especially all this data is already in the `head` or other schema used in the SEO include.

The `page.link` variable stuff is for “link” posts so you can’t rely on it. https://mmistakes.github.io/minimal-mistakes/post%20formats/post-link/

Maybe the best path forward is add the bare minimum of microformats that doesn’t duplicate markup. Then let me noodle on future support.

The theme is showing it’s age and a lot of other features being requested could benefit from me doing a complete overhaul. Microformats could be lumped in with that…

Reply:
Yes, I am disappointed that the IndieWeb folks aren't using something more structured like Schema.org, but I don't have any leverage to steer that community...and it seems like they have quite the organic mix of tool providers that are using microformats.

Are you okay with merging the pull request as it is now?  The minimal parts of microformats for Card and Entry have been added to the existing theme markup with the exception of the hidden &lt;ul&gt; list with the one list item for the URL to the post.  I could clean up the wording of the list item explaining the hidden list—say, "This list item holds the microformat entry for the link to this post, and it is hidden from view"—and make a final commit to this pull request, if you prefer.

Reply:
The hidden content still gives me a bad feeling.
In fact was looking at the Microformats wiki and they even discourage it.

> If some content doesn't fit into your visible design, then it's not worth microformatting it.

https://microformats.org/wiki/faq#Hidden_Content

Reply:
You're right: the Microformats wiki does discourage the use of hidden content.  That said, I think having something in the markup that points to itself seems to be a pretty critical idea—especially if some machine process is going to extract the data out of the page to make use of somewhere else.

I went poking through the Schema-org markup for [`CreativeWork`](https://schema.org/CreativeWork), and there is a property from the inherited `Thing` for URL—and that property isn't used in the mmistakes markup at the moment.  Perhaps a permalink can be added to the footer of the article, and that can be marked up with both `itemprop="headline"` (for Schema-org) and `class="u-url"` (for Microformats).

<img width="685" alt="image" src="https://user-images.githubusercontent.com/152960/126694680-24f49bf9-0662-4cbf-a4d7-25a31531dd2c.png">

```html
<p>
  <strong><i class="fas fa-fw fa-bullseye" aria-hidden="true"></i> Permalink:</strong> <span itemprop="url" class="u-url">https://dltj.org/article/dltj-with-webmention/</span>
</p>
```

If you like, I could wrap it in a Jekyll {% if %} block that checks for a feature flag that is set to "false" by default so it doesn't have a visible impact on existing sites.

Reply:
What if instead of adding it to the footer metas... and adding yet another config toggle, you include the permalink in the `h1` with the post title? I don't think it harms anything doing that.

```liquid
<h1 id="page-title" class="page__title" itemprop="headline">
  <a href="{{ page.url | absolute_url }}" class="u-url">{{ page.title | markdownify | remove: "<p>" | remove: "</p>" }}</a>
</h1>
```

About the only change would be a visual one as the title is now styled like a link... which I don't see any issue with.

![Screen Shot 2021-07-22 at 3 36 39 PM](https://user-images.githubusercontent.com/1376749/126699249-cb86205c-9f68-4952-ac87-b39e11044496.png)